Userland/Libraries/LibWeb/Layout/LineBox.cpp

```diff
@@ -42,6 +42,11 @@ void LineBox::trim_trailing_whitespace()
         return;
 
     auto& last_fragment = m_fragments.last();
+
+    auto* dom_node = last_fragment.layout_node().dom_node();
+    if (dom_node && dom_node->parent() && dom_node->parent()->is_html_pre_element())
+        return;
```

Comment:
`<pre>` elements should not be magical like this. Instead we should look at the CSS `white-space` property and choose behavior based on that :)

Reply:
Thanks, I sloppily didn't actually check how `<pre>` works, just assumed it was special :).

I fixed it now to behave according to the white-space property. 

However, it is a simple fix in accordance to the CSS2.1 spec. I also read the upcoming css-text-3 module, which has a more complicated handling. I did not trust myself though to implement that one, since it would require some bigger refactoring and restructuring of the current handling of justification, white-space collapse, alignment and transforms so that it's closer to that newer spec [1]

[1] https://www.w3.org/TR/css-text-3/#order